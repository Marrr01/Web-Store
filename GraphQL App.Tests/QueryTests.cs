using CookieCrumble;
using Xunit;

namespace GraphQL_App.Tests
{
    public class QueryTests
    {
        [Fact]
        public async Task GetUsers_CorrectQuery_ReturnsProductsInUsersBaskets()
        {
            await using var result = await TestServices.ExecuteRequestAsync(b => b.SetQuery(
                """
                query {
                  users {
                    login
                    password
                    baskets {
                      basketProducts {
                        productAmount
                        product {
                          name
                          price
                        }
                      }
                    }
                  }
                }
                """));
            result.MatchSnapshot();
        }
    }
}