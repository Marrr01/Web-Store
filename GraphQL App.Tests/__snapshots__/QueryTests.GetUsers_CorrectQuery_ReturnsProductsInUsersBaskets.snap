{
  "data": {
    "users": [
      {
        "login": "login 1",
        "password": "password",
        "baskets": [
          {
            "basketProducts": [
              {
                "productAmount": 1,
                "product": {
                  "name": "Product 1",
                  "price": 10
                }
              },
              {
                "productAmount": 2,
                "product": {
                  "name": "Product 2",
                  "price": 20
                }
              }
            ]
          },
          {
            "basketProducts": [
              {
                "productAmount": 1,
                "product": {
                  "name": "Product 1",
                  "price": 10
                }
              },
              {
                "productAmount": 2,
                "product": {
                  "name": "Product 2",
                  "price": 20
                }
              }
            ]
          }
        ]
      },
      {
        "login": "login 2",
        "password": "password",
        "baskets": [
          {
            "basketProducts": [
              {
                "productAmount": 1,
                "product": {
                  "name": "Product 1",
                  "price": 10
                }
              },
              {
                "productAmount": 2,
                "product": {
                  "name": "Product 2",
                  "price": 20
                }
              }
            ]
          },
          {
            "basketProducts": [
              {
                "productAmount": 1,
                "product": {
                  "name": "Product 1",
                  "price": 10
                }
              },
              {
                "productAmount": 2,
                "product": {
                  "name": "Product 2",
                  "price": 20
                }
              }
            ]
          }
        ]
      }
    ]
  }
}